src/goto-programs/builtin_functions.cpp

```diff
@@ -636,6 +636,88 @@ void goto_convertt::do_enum_is_in_range(
   copy(assignment, ASSIGN, dest);
 }
 
+void goto_convertt::do_havoc_slice(
+  const exprt &lhs,
+  const symbol_exprt &function,
+  const exprt::operandst &arguments,
+  goto_programt &dest)
+{
+  std::string identifier = CPROVER_PREFIX "havoc_slice";
+
+  // We disable checks on the generated instructions
+  // because we add our own rw_ok assertion that takes size into account
+  auto source_location = function.find_source_location();
+  source_location.add_pragma("disable:pointer-check");
+  source_location.add_pragma("disable:pointer-overflow-check");
+  source_location.add_pragma("disable:pointer-primitive-check");
+
+  // check # arguments
+  if(arguments.size() != 2)
+  {
+    error().source_location = source_location;
+    error() << "'" << CPROVER_PREFIX "havoc_slice"
+            << "' expected to have two arguments" << eom;
+    throw 0;
+  }
+
+  // check argument types
+  if(arguments[0].type().id() != ID_pointer)
+  {
+    error().source_location = source_location;
+    error() << "'" << CPROVER_PREFIX "havoc_slice"
+            << "' first argument expected to have `void *` type" << eom;
+    throw 0;
+  }
+
+  if(arguments[1].type().id() != ID_unsignedbv)
+  {
+    error().source_location = source_location;
+    error() << "'" << CPROVER_PREFIX "havoc_slice"
+            << "' second argument expected to have `size_t` type" << eom;
+    throw 0;
+  }
+
+  // check nil lhs
+  if(lhs.is_not_nil())
+  {
+    error().source_location = source_location;
+    error() << "'" << identifier << "' not expected to have a LHS" << eom;
+    throw 0;
+  }
+
+  // insert instructions
+  // assert(rw_ok(argument[0], argument[1]));
+  // char ndarr[argument[1]];
+  // __CPROVER_array_replace(p, ndarr);
+
+  r_or_w_ok_exprt ok_expr(ID_w_ok, arguments[0], arguments[1]);
+  ok_expr.add_source_location() = source_location;
+  goto_programt::targett t =
+    dest.add(goto_programt::make_assertion(ok_expr, source_location));
+  t->source_location_nonconst().set("user-provided", false);
+  t->source_location_nonconst().set_property_class(ID_assertion);
+  t->source_location_nonconst().set_comment(
+    "assertion havoc_slice " + from_expr(ns, identifier, ok_expr));
```

Comment:
Rather than fixing up the location after the fact, make these changes to a source location beforehand, like:
````c++
source_locationt annotated_location = source_location;
annotated_location.set("user-provided", ...); ...
````
and then use `annotated_location` when constructing the goto-program instruction.

Reply:
done

Reply:
Did you perhaps end up going back and forth on this? It doesn't seem to have been addressed.